src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -77,11 +77,15 @@ protected Signature(SignerInfo signerInfo, SignatureType type)
                 throw new ArgumentNullException(nameof(issues));
             }
 
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
```

Comment:
style: you can inline this will null coalescing. 

I'm sure the analyzers recommend this.